WikiSearch.java

```diff
@@ -0,0 +1,311 @@
+
+                        if (val1 == val2) {
+                                return 0;
+                        }
+                        else if (val1 < val2) {
+                                return 1;
+                        }
+                        else {
+                                return -1;
+                        }
+                }
+        };
+
+
+	/**
+	 * Constructor.
+	 * 
+	 * @param map
+	 */
+	public WikiSearch(Map<String, Integer> tfMap) {
+		this.tfMap = tfMap;
+	}
+
+	public WikiSearch(Map<String, Double> tfidfMap, String term) {
+		this.tfidfMap = tfidfMap;
+	}
+
+	public WikiSearch(Map<String, Integer> tfMap, Map<String, Double> tfidfMap) {
+		this.tfMap = tfMap;
+		this.tfidfMap = tfidfMap;
+	}
+	
+	public WikiSearch() {
+		this.tfMap = new HashMap<String, Integer>();
+		this.tfidfMap = new HashMap<String, Double>();
+	}
+
+	/**
+	 * Looks up the relevance of a given URL.
+	 * 
+	 * @param url
+	 * @return
+	 */
+	public Integer getRelevance(String url) {
+		Integer relevance = tfMap.get(url);
+		return relevance==null ? 0: relevance;
+	}
+	
+	/**
+	 * Prints the contents in order of term frequency.
+	 * 
+	 * @param map
+	 */
+	public void print() {
+		List<Entry<String, Integer>> entries = sort();
+		if (entries.isEmpty()) {
+			System.out.println("No relevant pages found");
+		}
+		else {
+			for (Entry<String, Integer> entry: entries) {
+				System.out.println(entry);
+			}
+		}
+	}
+
+	public void printIDF() {
+		if (tfMap.entrySet().isEmpty()) {
+			System.out.println("No relevant pages found");	
+		}
+		else {
+			List<Entry<String, Double>> entries = sortIDF();
+			for (Entry<String, Double> entry: entries) {
+				if (entry.getValue() == 0) {
+					System.out.println("Query not specific enough");	
+					break;
+				}
+				else {
+					System.out.println(entry);
+				}
+			}
+		}
+	}
+	
+	/**
+	 * Computes the union of two search results.
+	 * 
+	 * @param that
+	 * @return New WikiSearch object.
+	 */
+	public WikiSearch or(WikiSearch that) {
+        // FILL THIS IN!
+		Map<String, Integer> newMap = new HashMap<String, Integer>();
+		newMap.putAll(tfMap);
+
+		for (String url: that.tfMap.keySet()) {
+			if (newMap.containsKey(url)) {
+				Integer orRelevance = that.tfMap.get(url) + this.tfMap.get(url);
+				newMap.put(url, orRelevance);
+			}
+			else {
+				newMap.put(url, that.tfMap.get(url));
+			}
+		}
+
+		Map<String, Double> newMapIDF = new HashMap<String, Double>();
+		newMapIDF.putAll(tfidfMap);
+
+                for (String url: that.tfidfMap.keySet()) {
+                        if (newMapIDF.containsKey(url)) {
+                                Double orRelevance = that.tfidfMap.get(url) + this.tfidfMap.get(url);
+                                newMapIDF.put(url, orRelevance);
+                        }
+                        else {
+                                newMapIDF.put(url, that.tfidfMap.get(url));
+                        }
+                }
+		
```

Comment:
I'm pretty confident you cannot just add the tf scores and tfidf scores to implement the OR operator or the AND operator. This simply makes no sense and will give nonsense results. The fact that your OR and AND methods have the same implementation should be a red flag that both are incorrect. Normally, when performing queries with multiple terms you use some kind of vector space model, like the one discussed here: https://www.elastic.co/guide/en/elasticsearch/guide/current/scoring-theory.html

Even the vector space model doesn't support boolean queries as such, just multi-term queries. Without delving into really complex models, relevance models like tf-idf are generally incompatible with boolean queries. I have no idea what they expect you to do here, but I'm guessing they're expecting you to implement simple boolean queries, where you simply return all documents that match the boolean query in no particular order. For this, you do not need to and shouldn't track relevance scores like tf and tfidf.